### ISSUE TITLE
[Feature request] read key config from eeprom

### ISSUE THREAD

**Issue by biemster** [wagiminator/CH552-Macropad-mini#1]
The ch552 has a guaranteed (per datasheet...) firmware write lifetime of 200 writes. This is quite little, and for experimenters easy to pass in a couple days. If the key config (which characters are send over USB when a key is pressed) could be read from the internal eeprom, at least those experiments would not count to the flash write wear.
I'm going to implement this for my macropad (https://hackaday.io/project/189914-rgb-macropad-custom-firmware) so there will be a PR for this at a certain point, but it might take a while due to freetime constraints.

**Comment by wagiminator** [comment]
Yes, 200 write cycles is really little for developers. The use of the data flash is therefore actually useful and can also be implemented without any problems for most key combinations. It may need an additional interface (maybe CDC) to upload the macros. It will probably become problematic for more complex macros that consist of several key combinations and key sequences with delays in between. I am looking forward to your implementation and wish you every success. Let's see if I can implement something similar for my MacroPad in the future, but for now there are other projects on my agenda.
